[Dependency Cache] Enable Dependency Cache on CI [without GRADLE_RO_DEP_CACHE]

ParaskP7 opened the pull request:
Related: #21540

## Description:

This is a test only PR that attempts to test another dependency cache mechanism, one that does NOT dependent on the read-only cache ([GRADLE_RO_DEP_CACHE](https://docs.gradle.org/current/userguide/dependency_caching.html#sec:shared-readonly-cache)), and per project.

-----

## To Test:

<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->

`N/A`

> [!TIP]
> To verify that `GRADLE_RO_DEP_CACHE` is **NOT** enabled make sure you don't see this `Shared read-only dependency cache is an incubating feature.` log when Gradle launches for a CI build ([example](https://buildkite.com/automattic/wordpress-android/builds/20727#0193df28-3e2a-46e6-98e9-166835d11945/226-243)).

-----

## Regression Notes: `N/A`

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones): `N/A`

ParaskP7 committed:
CI: Use android-staging agent to test with ro dependency cache disabled

AMI ID: ami-06a633c14b35c40b9
AMI Name: android-build-image-6.12.0v1.4-rc-1

ParaskP7 committed:
CI: Add a scheduled dependency cache job

FYI: This job will be then used by 'buildkite-ci' and configured as a
'buildkite_pipeline_schedule' with a weekly frequency.

ParaskP7 committed:
CI: Add restore cache command to run on targeted steps

The targeted steps are:
- Mobile Apps
- Lint
- Detekt
- CheckStyle
- Unit Tests
- Android tests

ParaskP7 committed:
CI: Use non read-only dependency cache to save/restore the cache

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ParaskP7 committed:
CI: Fix no such file or directory type of errors when using multi-cmd

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into ci/enable-dependency-cache-on-ci-v2

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into ci/enable-dependency-cache-on-ci-v2

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into ci/enable-dependency-cache-on-ci-v2

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into ci/enable-dependency-cache-on-ci-v2

ParaskP7 committed:
CI: Trigger a test build using the latest update from a8c-ci-toolkit

PR: https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/
pull/135

ParaskP7 committed:
CI: Switch back to android agent to re-test without ro dependency cache

After deploying this buildkite-ci#570 change (PR below), which removed
the read-only dependency cache (GRADLE_RO_DEP_CACHE), the agent can be
switched back to 'android' for us to test that the deployment went as
expected, just to be sure.

PR: https://github.com/Automattic/buildkite-ci/pull/570

AMI ID: ami-0fdecc22b38a48860
AMI Name: android-build-image-6.12.0v1.4

ParaskP7 committed:
Merge branch 'trunk' of github.com:wordpress-mobile/WordPress-Android into ci/enable-dependency-cache-on-ci-v2
